Cron/EventsTopic.php

```diff
@@ -0,0 +1,190 @@
+        SyncsFactory $klSyncFactory,
+        QuoteIdMask $quoteIdMaskResource,
+        CategoryFactory $categoryFactory
+    )
+    {
+        $this->_klaviyoLogger = $klaviyoLogger;
+        $this->_eventsCollectionFactory = $eventsCollectionFactory;
+        $this->_klSyncFactory = $klSyncFactory;
+        $this->_quoteIdMaskResource = $quoteIdMaskResource;
+        $this->_categoryFactory = $categoryFactory;
+    }
+
+    /**
+     * Gets 500 rows from the kl_events table and move to the kl_sync table
+     */
+    public function moveRowsToSync()
+    {
+        // New Events to be moved to kl_sync table and update status of these to Moved, limit 500
+        $eventsCollection = $this->_eventsCollectionFactory->create();
+        $eventsData = $eventsCollection->getRowsForSync('NEW')
+            ->addFieldToSelect(['id','event','payload','user_properties'])
+            ->getData();
+
+        if (empty($eventsData)){
+            return;
+        }
+
+        $idsMoved = [];
+
+        // Capture all events that have been moved and add data to Sync table
+        foreach ($eventsData as $event){
+            if ($event['event'] == 'Added To Cart') {
+                $event['payload'] = json_encode($this->replaceQuoteIdAndCategoryIds($event['payload']));
+            }
+
+            //TODO: This can probably be done as one bulk update instead of individual inserts
+            $sync = $this->_klSyncFactory->create();
+            $sync->setData([
+                'status' => 'NEW',
+                'topic' => $event['event'],
+                'user_properties' => $event['user_properties'],
+                'payload' => $event['payload']
+            ]);
+            try {
+                $sync->save();
+                array_push($idsMoved, $event['id']);
+            } catch (\Exception $e) {
+                $this->_klaviyoLogger->log(sprintf("Unable to move row: %s", $e->getMessage()));
+            }
+        }
+
+        // Update Status of rows in kl_events table to Moved
+        $eventsCollection->updateRowStatus($idsMoved, 'MOVED');
+    }
+
+    /**
+     * Deletes rows moved to the kl_syncs table that are older than 2 days
+     */
+    public function deleteMovedRows()
+    {
+        // Delete rows that have been moved to sync table
+        $eventsCollection = $this->_eventsCollectionFactory->create();
+        $idsToDelete = $eventsCollection->getIdsToDelete('MOVED');
+
+        $eventsCollection->deleteRows($idsToDelete);
+    }
+
+    /**
+     * Helper function to replace QuoteId with MaskedQuoteId and CategoryIds with CategoryNames in Added To Cart payload
+     * @param string $payload
+     * @return array
+     */
+    public function replaceQuoteIdAndCategoryIds(string $payload): array
+    {
+        $decoded_payload = json_decode($payload, true);
+        $maskedQuoteId = $this->_quoteIdMaskResource->getMaskedQuoteId($decoded_payload['QuoteId']);
+        $decoded_payload['MaskedQuoteId'] = $maskedQuoteId;
```

Comment:
the masked quote id is available only for guest carts AFAIK, so probably it will just remove the quote id for registered customers